lib/tree.rb

```diff
@@ -104,22 +120,48 @@ def self.find_tree(repository, file_path, revstring=Constants::HEAD)
 
     private
 
-    def jtree_entries(options={})
+    def contents_array
+      @contents_ary ||= jtree_entries
+    end
+
+    def init_walk(recurse=false)
       treewalk = TreeWalk.new(@jrepo)
       treewalk.add_tree(@jtree)
-      treewalk.set_recursive(true) if options[:recursive]
+      treewalk.set_recursive(true) if recurse
+      treewalk
+    end
+
+    def tree_entry(treewalk)
+      mode = treewalk.get_file_mode(0)
+      {
```

Comment:
Stylistically, I would expect the opening bracket to follow the `get_file_mode(0)` call (and not be on a separate line).

Reply:
This is not the opening bracket of a block but of a results hash :)